File: sdkwork-ai-sdk-python/sdkwork_ai_sdk/api/files2.py
from typing import Any, Dict, List, Optional
from ..http_client import HttpClient
from ..models import OpenAiFileDeleteResponse, OpenAiFileList, OpenAiFileObject, UploadFilePostRequest

class Files2Api:
    """Files API client."""
    
    def __init__(self, client: HttpClient):
        self._client = client

    def list_files(self, params: Optional[Dict[str, Any]] = None) -> OpenAiFileList:
        """List files"""
        return self._client.get(f"/ai/v3/v1/files", params=params)

    def upload_file(self, body: Optional[UploadFilePostRequest] = None, params: Optional[Dict[str, Any]] = None) -> OpenAiFileObject:
        """Upload file"""
        return self._client.post(f"/ai/v3/v1/files", data=body, params=params)

    def retrieve_file_content(self, fileId: str) -> str:
        """Retrieve file content"""
        return self._client.get(f"/ai/v3/v1/files/{fileId}/content")

    def retrieve_file(self, fileId: str) -> OpenAiFileObject:
        """Retrieve file"""
        return self._client.get(f"/ai/v3/v1/files/{fileId}")

    def delete_file(self, fileId: str) -> OpenAiFileDeleteResponse:
        """Delete file"""
        return self._client.delete(f"/ai/v3/v1/files/{fileId}")

File: sdkwork-ai-sdk-flutter/lib/src/api/files2.dart
import '../http/client.dart';
import '../models.dart';

class Files2Api {
  final HttpClient _client;
  
  Files2Api(this._client);

  /// List files
  Future<OpenAiFileList?> listFiles(Map<String, dynamic>? params) async {
    final response = await _client.get(ApiPaths.aiPath('/v1/files'), params: params);
    return response is OpenAiFileList ? response : null;
  }

  /// Upload file
  Future<OpenAiFileObject?> uploadFile(UploadFilePostRequest? body, Map<String, dynamic>? params) async {
    final response = await _client.post(ApiPaths.aiPath('/v1/files'), body: body, params: params, contentType: 'multipart/form-data');
    return response is OpenAiFileObject ? response : null;
  }

  /// Retrieve file content
  Future<String?> retrieveFileContent(String fileId) async {
    final response = await _client.get(ApiPaths.aiPath('/v1/files/${fileId}/content'));
    return response is String ? response : null;
  }

  /// Retrieve file
  Future<OpenAiFileObject?> retrieveFile(String fileId) async {
    final response = await _client.get(ApiPaths.aiPath('/v1/files/${fileId}'));
    return response is OpenAiFileObject ? response : null;
  }

  /// Delete file
  Future<OpenAiFileDeleteResponse?> deleteFile(String fileId) async {
    final response = await _client.delete(ApiPaths.aiPath('/v1/files/${fileId}'));
    return response is OpenAiFileDeleteResponse ? response : null;
  }
}
